main/zoe/guide/contracts/atomic-swap.md

```diff
@@ -2,130 +2,80 @@
 ```
-
-In order for Alice to escrow with Zoe she needs to use her invite.
-Once Alice uses her invite and makes her offer she will receive an `outcome` and a
-promise that resolves to her payout.
-
+Next, we use `zcf.MakeInvitation()` to make an `invite` to send to Bob. The
+`invite` is a `Payment` minted from Zoe's internal `inviteMint` This
+invite includes an offer, so our first `offerhook` argument will be invoked in the
+contract with an `offerHandle`. The `offerhook`'s result is returned as the outcome
+of making the offer via the invitation.
+
+Here, our `offerHook` uses the powerful `swap()` ZoeHelper. It takes
+two offers which satisfy each other's `want` and `give` values and then
+makes a swap between them. Zoe reallocates their assets, and completes both
+offers.
+
+The second `inviteDesc` argument `'matchOffer'` is a string that describes the `invite` for search purposes.
+The third `customProperties` argument is an object whose properties are information needed
+by the contract and included in the invitations's extent (recall that `Payments` are `amounts`, which 
+have `extent` and `brand` components).
 ```js
-const { outcome, payout: alicePayoutP } = await E(zoe).offer(
-  aliceInvite,
-  aliceProposal,
-  alicePayments,
-);
+return zcf.makeInvitation(
+      offerHandle => swap(firstOfferHandle, offerHandle),
+      'matchOffer',
+      harden({
+        customProperties: {
+          asset: give.Asset,
+          price: want.Price,
+        },
+      }),
+    );
+  };
 ```
-
-Alice's outcome, the result of the offer, is an invite she can send to
-someone else:
-
-```js
-const newInviteP = outcome;
-```
-
-She then sends the invite to Bob and he looks up the invite to see if it matches Alice's claims.
-
-```js
-const inviteIssuer = zoe.getInviteIssuer();
-const bobExclusiveInvite = await inviteIssuer.claim(newInviteP);
-const bobInviteExtent = inviteIssuer.getAmountOf(bobExclusiveInvite)
-  .extent[0];
-
-const {
-  installationHandle: bobInstallationId,
-  issuerKeywordRecord: bobIssuers,
-} = zoe.getInstance(bobInviteExtent.instanceHandle);
-
-
-// Bob does checks
-assert(bobInstallationId === installationHandle, details`wrong installation`);
-assert(bobIssuers.Asset === moolaIssuer, details`unexpected Asset issuer`);
-assert(bobIssuers.Price === simoleanIssuer, details`unexpected Price issuer`);
-assert(moolaAmountMath.isEqual(bobInviteExtent.asset, moola(3)), details`wrong asset`);
-assert(simoleanAmountMath.isEqual(bobInviteExtent.price, simoleans(7)), details`wrong price`);
+Here we define the offer format for this contract.
+ ```js 
+  const firstOfferExpected = harden({
+    give: { Asset: null },
+    want: { Price: null },
+  });
 ```
-
-Bob decides to be the counter-party. He also escrows his payments and uses his invite to
-make an offer in the same way as Alice, but his `Proposal` match Alice's:
-
+And finally, we return our invitation to send to Bob. **tyg todo: Not clear on why there are two makeInvitation calls?**
```

Comment:
There are two makeInvitation calls because Alice needs an invite and Bob needs a different invite.